xml/security_ldap.xml

```diff
@@ -398,266 +399,195 @@ objectclass (2.16.840.1.113730.3.2.333 NAME 'nsPerson' <co xml:id="co-ldap-schem
 
 [slapd]
-root_password = <replaceable>YOUR_PASSWORD_FOR_CN=DIRECTORY_MANAGER</replaceable><co xml:id="co-ldap-389-ds-rootpasswd"/>
+root_password = <replaceable>password</replaceable><co xml:id="co-ldap-389-ds-rootpasswd"/>
+self_sign_cert = True <co xml:id="co-ldap-389-ds-cert"/>
+instance_name = <replaceable>389ds-test</replaceable>
 
 [backend-userroot]
-sample_entries = yes
-suffix = dc=example,dc=com</screen>
+sample_entries = yes <co xml:id="co-ldap-389-ds-sample-entries"/>
+suffix = <replaceable>dc=test1,dc=com</replaceable></screen>
+
       <calloutlist>
+       <callout arearefs="co-ldap-389-ds-config_version">
+        <para>
+         This line is required, indicating that this is a version 2 setup .inf
+         file.
+        </para>
+       </callout>
        <callout arearefs="co-ldap-389-ds-rootpasswd">
         <para>
-         Set the <varname>root_password</varname> to the password for
-         the directory server &rootuser; user. The password is used for LDAP
-         server administration only.
+         Create a <varname>root_password</varname> for
+         the directory server &rootuser; user, <literal>dirsrv</literal>. 
+         The <literal>dirsrv</literal> user has no login or shell privileges, 
+         and is used for LDAP server administration only.
+        </para>
+       </callout>
+       <callout arearefs="co-ldap-389-ds-cert">
+        <para>
+         Create self-signed server certificates in 
+         <filename>/etc/dirsrv</filename>.
         </para>
        </callout>
+       <callout arearefs="co-ldap-389-ds-sample-entries">
+        <para>
+         Populate the new instance with sample user and group entries.
+        </para>
+       </callout>       
       </calloutlist>
      </example>
    </step>
    <step>
     <para>
      To create the &ds389; instance from <xref linkend="ex-ldap-389-ds-inf"
-     xrefstyle="select:label"/>, run:
+     xrefstyle="select:label"/>, run the following command:
     </para>
-    <screen>&prompt.sudo;dscreate from-file /tmp/instance.inf</screen>
+    <screen>&prompt.sudo;dscreate -v from-file 389ds-test.inf | tee 389ds-test-output.txt</screen>
     <para>
-     This creates a working LDAP server.
+     This shows all activity during the instance creation, stores all the messages in
+     <filename>389ds-test-output.txt</filename>, and creates a working 
+     LDAP server in about a minute. The verbose output contains a lot of useful
+     information. If you don't want to save it then delete the 
+     <literal>| tee 389ds-test-output.txt</literal> portion of the command.
     </para>
    </step>
    <step>
     <para>
-     If <command>dscreate</command> should fail, the messages will tell you why.
-     For more details, repeat the command with the <option>-v</option> option:
+     If the <command>dscreate</command> command should fail, the messages will 
+     tell you why. After correcting any issues, re-run the command.    
```

Comment:
I think after a failure, you actually need to do a `dsctl <instance name> remove` before you can re-run. That's because during the setup the *very* first thing we do is create an instance marker file that says 'yeah I'm here! I'm real!'. So any setup "after" will see the marker and say "lol soz im not gonna install, this already exists".

The removal tool is able to remove even a partially installed instance, so that at "any" stage of the install, any nature of failure can be cleaned up, where even if the remover fails, it can be re-run until it suceeds at the removal. Only at the very end of removal do we remove the marker file as the very last action to say "yes this really does no longer exist".

Anyway, that was a long way to say "if you have a problem, remove THEN recreate". But the process is really carefully made so that that works no matter the state of the install.

Reply:
:+1: